torch/csrc/distributed/rpc/script_functions.cpp

```diff
@@ -0,0 +1,55 @@
+#include <torch/csrc/distributed/rpc/script_functions.h>
+
+#include <torch/csrc/distributed/autograd/utils.h>
+#include <torch/csrc/distributed/rpc/message.h>
+#include <torch/csrc/distributed/rpc/python_rpc_handler.h>
+#include <torch/csrc/distributed/rpc/rpc_agent.h>
+#include <torch/csrc/distributed/rpc/script_call.h>
+#include <torch/csrc/distributed/rpc/utils.h>
+
+namespace torch {
+namespace distributed {
+namespace rpc {
+
+c10::intrusive_ptr<c10::ivalue::Future> rpcTorchscriptCall(
+    const std::string& dst,
+    const c10::QualifiedName& qualifiedName,
+    std::vector<c10::IValue>& stack) {
+  auto scriptCall =
+      std::make_unique<ScriptCall>(qualifiedName, std::move(stack));
+  auto agent = RpcAgent::getDefaultRpcAgent();
+  auto futMessage = autograd::sendMessageWithAutograd(
+      *agent, agent->getWorkerInfo(dst), std::move(*scriptCall).toMessage());
+  // Get function return type to construct c10::ivalue::Future.
+  // Script call only allows single IValue returned.
+  auto returns = PythonRpcHandler::getInstance()
+                     .jitCompilationUnit()
+                     ->get_function(qualifiedName)
```

Comment:
I think you have looked up for this function outside of "rpcTorchscriptCall", can we pass the function as a reference to this function?

Reply:
rpcTorchscriptCall will be called in JIT later on, so the signature of rpcTorchscriptCall should be compitable with JIT ops, that can not accept function as arg